src/v/cluster/members_manager.cc

```diff
@@ -501,6 +501,37 @@ members_manager::dispatch_configuration_update(model::broker broker) {
     }
 }
 
+bool is_result_configuration_valid(
+  const std::vector<broker_ptr>& current_brokers,
+  const model::broker& to_update) {
+    /**
+     * validate if any two of the brokers would listen on the same addresses
+     * after applying configuration update
+     */
+    for (auto& current : current_brokers) {
+        if (current->id() == to_update.id()) {
+            continue;
+        }
+        // error, nodes would point to the same addresses
+        if (current->rpc_address() == to_update.rpc_address()) {
+            return false;
+        }
+        for (auto& current_ep : current->kafka_advertised_listeners()) {
+            auto any_is_the_same = std::any_of(
+              to_update.kafka_advertised_listeners().begin(),
+              to_update.kafka_advertised_listeners().end(),
+              [&current_ep](const model::broker_endpoint& ep) {
+                  return current_ep == ep;
+              });
```

Comment:
is this an overly restrictive constraint given that kafka advertised listeners are not involved in the correctness of any low-level raft bits?

Reply:
I think not. Having two advertised addresses pointing to the same listener may cause invalid client behavior. f.e. client will always receive `not_leader_for_partition` error